### PR TITLE
Metrics terminology (Adding/Grouping)

### LINKED ISSUE #625: Metrics terminology: Grouping instruments as (opposed to Adding instruments)
The current metrics spec uses the term "Additive" to describe the Counter, UpDownCounter, SumObserver, and UpDownSumObserver instruments. This term was derived as follows: the input to these instruments convey a sum that is of primary interest, whether expressed as the change (synchronous case) or as the sum itself (synchronous case).

The other instruments have not had a proper term applied.  Sometimes "non-additive", these instruments recognize individual events cannot be simply added, that they are averaged instead of added. This proposal suggests the term _Collective_ to refer to these instruments, because:

- It has the same adjective suffix "-ive", meaning "[that performs or tends toward an (indicated) action](https://www.merriam-webster.com/dictionary/-ive)"
- Sense of meaning "[of, relating to, or being a group of individuals](https://www.merriam-webster.com/dictionary/collective)"

Examples of use:
- Collective instruments generally cost more than Additive instruments
- Collective instruments are used to capture a distribution of values
- The default aggregator for Collective instruments is MinMaxSumCount
- The legacy term "Gauge" usually implies a Collective instrument
- #608 proposes to add a new Collective instrument.

### PR DESCRIPTION
**Pull request by jmacd**
## Changes

Apply #625 decision to the API spec: The term "Additive" is replaced by "Adding", and "Non-Additive" is replaced by "Grouping". See the [comment here](https://github.com/open-telemetry/opentelemetry-specification/issues/625#issuecomment-676563191).

Fixes #625.

### CODE REVIEW

File: specification/metrics/api.md [hunk 1]
```diff
@@ -437,35 +437,35 @@ corresponding to the instrument and label set.  (For this reasons,
...
-In both synchronous and asynchronous cases, the additive instruments
+In both synchronous and asynchronous cases, the adding instruments
 are inexpensively aggregated into a single number per collection interval
-without loss of information.  This property makes additive instruments
-higher performance, in general, than non-additive instruments.
+without loss of information.  This property makes adding instruments
+higher performance, in general, than grouping instruments.
 
-Non-additive instruments use a relatively inexpensive aggregation
+grouping instruments use a relatively inexpensive aggregation
```
Comment (jkwatson) [inline specification/metrics/api.md:460]:
```suggestion
Grouping instruments use a relatively inexpensive aggregation
```

File: specification/metrics/api.md [hunk 2]
```diff
@@ -110,24 +110,24 @@ has put effort into taking some kind of measurement.  For both
...
 
-The term _additive_ is used to specify a characteristic of some
+The term _adding_ is used to specify a characteristic of some
 measurements, meant to indicate that only the sum is considered useful
 information.  These are measurements that you would naturally combine
 using arithmetic addition, usually real quantities of something
 (e.g., number of bytes).
 
-Non-additive measurements are used when the set of values, also known
+grouping measurements are used when the set of values, also known
```
Comment (jkwatson) [inline specification/metrics/api.md:119]:
```suggestion
Grouping measurements are used when the set of values, also known
```

File: specification/metrics/api.md [hunk 3]
```diff
@@ -110,24 +110,24 @@ has put effort into taking some kind of measurement.  For both
...
 measurement you would naturally add where the intention is to monitor
 the distribution of values (e.g., queue size).  The median value is
-considered useful information for non-additive measurements.
+considered useful information for grouping measurements.
 
-Non-additive instruments semantically capture more information than
-additive instruments.  Non-additive measurements are more expensive
-than additive measurements, by this definition.  Users will choose
-additive instruments except when they expect to get value from the
+grouping instruments semantically capture more information than
```
Comment (jkwatson) [inline specification/metrics/api.md:127]:
```suggestion
Grouping instruments semantically capture more information than
```

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

File: specification/metrics/api.md [hunk 4]
```diff
@@ -110,24 +110,24 @@ has put effort into taking some kind of measurement.  For both
...
 the distribution of values (e.g., queue size).  The median value is
-considered useful information for non-additive measurements.
+considered useful information for grouping measurements.
 
-Non-additive instruments semantically capture more information than
-additive instruments.  Non-additive measurements are more expensive
-than additive measurements, by this definition.  Users will choose
-additive instruments except when they expect to get value from the
+Grouping instruments semantically capture more information than
+adding instruments.  grouping measurements are more expensive
```
Comment (arminru) [inline specification/metrics/api.md:128]:
```suggestion
adding instruments.  Grouping measurements are more expensive
```

File: specification/metrics/api.md [hunk 5]
```diff
@@ -149,13 +149,13 @@ Instruments are classified in several ways that distinguish them from
 one another.
 
 1. Synchronicity: A synchronous instrument is called by the user in a distributed [Context](../context/context.md) (i.e., Span context, Baggage). An asynchronous instrument is called by the SDK once per collection interval, lacking a Context.
-2. Additivity: An additive instrument is one that records additive measurements, as described above.
-3. Monotonicity: A monotonic instrument is an additive instrument, where the progression of each sum is non-decreasing.  Monotonic instruments are useful for monitoring rate information.
+2. Adding vs. Grouping: An adding instrument is one that records adding measurements, as opposed to a grouping instrument as described above.
```
Comment (arminru) [inline specification/metrics/api.md:152]:
```suggestion
2. Adding vs. Grouping: An adding instrument is one that records adding measurements, as opposed to a grouping instrument as described above.
```

**Review by arminru — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Comment by jmacd** [comment]
This is ready to merge.

**Pull request merged**